packages/fetch/src/fetch.ts

```diff
@@ -14,8 +22,33 @@ export function fetch(scope: Task, info: RequestInfo, init: RequestInit = {}): O
       }
     });
 
-    init.signal = controller.signal;
+    requestInit.signal = controller.signal;
+
+    let response = yield nativeFetch(info, requestInit);
+    return response;
+  };
 
-    return yield nativeFetch(info, init);
-  }
-};
+  return {
+    init,
+    arrayBuffer: function*() {
+      let response = yield { init };
+      return yield response.arrayBuffer();
+    },
+    blob: function*() {
+      let response = yield { init };
+      return yield response.blob();
+    },
+    formData: function*() {
+      let response = yield { init };
+      return yield response.formData();
+    },
+    json: function*() {
+      let response = yield { init };
+      return yield response.json();
+    },
+    text: function*() {
+      let response = yield { init };
+      return yield response.text();
+    },
```

Comment:
For these operations, would it make sense to curry so they are stateless? E.g.

```js
{
  json: () => function*() {}
}
```

That way you can consume it as a value:

```ts
export const status =  fetch('api.github.com/v2/status').json()
```

Otherwise, I'd just write them using method syntax:

```ts
{
  *json() {
     let response = yield { init };
     return yield response.json();
  }
}
```

Reply:
```
export const status =  fetch('api.github.com/v2/status').json()
```

This is not a pattern I'm a fan of encouraging, given that it depends on the implementation of `json` whether it'll work. Generally we should not be encouraging creating an operation which is shared among multiple spawn/yield points, even in those cases where that technically works.